app/controllers/sign_up/email_confirmations_controller.rb

```diff
@@ -31,7 +32,11 @@ def process_successful_confirmation
     end
 
     def store_sp_metadata_in_session
-      StoreSpMetadataInSession.new(session:, request_id:).call if request_id.present?
+      if request_id.present?
+        StoreSpMetadataInSession.new(session:, request_id:).call
```

Comment:
I'm curious if this should be happening in the `StoreSpMetadataInSession` class itself. Another place we use this is in "Forgot password" flow. Wondering if there's maybe a similar issue there.

Reply:
yea I was wondering if we wanted it there, but it was in multiple places including the IDV. But good call on the forgot password flow as well. It may be worth it in the IDV as well jic they switch computers in the middle of processing.

Reply:
If we move it into StoreSpMetadataInSession, we should take care to then remove the `bump_auth_count` from AuthorizationController and SamlIdpController because we call `StoreSpMetadatainsession` already in the those classes and would bump the auth_count if we move it to that concern

Reply:
Ugh Unfortunately I think doing it in the `StoreSpMetaDatainSession` causes issues since we call that service in the controllers that does the redirect and I think for now it might be easier to place it explicitly in areas we know we dont want the authorization confirmation to occur.